### PR TITLE
Refactored src/screens/UserPortal/Volunteer/Invitations from jest to vitest

### LINKED ISSUE #2583: Refactor: `src/screens/UserPortal/Volunteer/Invitations`  from Jest to Vitest
**Is your feature request related to a problem? Please describe.**
This file is currently tested using Jest. As part of our migration to Vitest, we need to refactor this file's test cases to align with Vitest's syntax and features.

**Acceptance Criteria**
1. Replace Jest-specific functions and mocks with Vitest equivalents
2. Ensure all tests in `src/screens/UserPortal/Volunteer/Invitations` pass after migration using `npm run test:vitest`
3. Maintain the test coverage for the file as 100% after migration
4. Upload a video or photo for this specific file coverage is 100% in the PR description

**Additional context**

- Refer to the merged PR for the base Vitest configuration:
https://github.com/PalisadoesFoundation/talawa-admin/pull/2457/files#diff-9a5a35cf8d7f02b2bba90905714b762560579c4789e9eb4cebb4607980aa3d93
- Vitest documentation:
https://vitest.dev/

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by syedali237**
**What kind of change does this PR introduce?**

Refactoring & Bugfix

**Issue Number:**

Fixes #2583 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

Screenshot 2024-12-20 at 7 11 22 PM

Screenshot 2024-12-20 at 7 11 10 PM

**If relevant, did you update the documentation?**
No

**Summary**

Refactored the test case to be run using vitest.

**Does this PR introduce a breaking change?**

No

**Other information**
Changed the tagTemplate.ts file, it was giving error while commiting updated the schema to the latest documentation.

Screenshot 2024-12-20 at 7 17 41 PM

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **New Features**
	- Enhanced structure of the tagTemplate for improved schema definition.

- **Bug Fixes**
	- Updated testing framework from Jest to Vitest, improving test management and mock handling.

- **Chores**
	- Adjusted string formatting in the basePdf property to comply with JSON standards.

### CODE REVIEW

**Comment by palisadoes** [comment]
Please fix:

1. Any failing tests (Click on the `details` link for more information)
1. Any conflicting files

Make sure CodeRabbit.ai approves your changes

**Comment by syedali237** [comment]
@palisadoes I have updated the template schema to align with the documentation provided by [pdfme](https://pdfme.com/).
After making the necessary adjustments to the schema, the typecheck now passes successfully when run locally (npm run typecheck shows no errors).

Screenshot 2024-12-21 at 4 52 07 PM

However, the GitHub action is still failing with the same type error, as shown below:

Screenshot 2024-12-21 at 4 46 52 PM

It’s unclear why this discrepancy is happening, as the error is no longer reproducible locally. Could this be related to differences in the CI environment, such as cached dependencies or an outdated @pdfme/common package? 

Any insights or suggestions to resolve this issue?


From the documentation below its clear that 'name' property does exist.

Screenshot 2024-12-21 at 4 39 25 PM

File: src/components/CheckIn/tagTemplate.ts [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
+        "width": 58.55,
+        "height": 5.67,
+        "alignment": "center",
+        "fontSize": 16,
+        "characterSpacing": 0,
+        "lineHeight": 1,
+        "fontName": "Roboto",
+        "fontColor": "#08780b",
+      }
+    ]
```
[the remark this answers is not in the document]
Comment (syedali237) [reply src/components/CheckIn/tagTemplate.ts:19]: The name property is defined in the @pdfme/common documentation and is valid for the schema. I’ve updated the schema to align with the documentation, and npm run typecheck now passes locally without any errors. GitHub CI failure might be due to cached dependencies or an outdated version of @pdfme/common in the CI environment.

[the remark this answers is not in the document]
Comment (syedali237) [reply src/components/CheckIn/tagTemplate.ts:19]: @coderabbitai if you could generate an interface?

**Comment by palisadoes** [comment]
Please fix the failing tests. Our test code coverage system will fail if any of these conditions occur:

1. The overall code coverage drops below the target threshold of the repository
2. Any file in the pull request has code coverage levels below the repository threshold
3. The code does not comply with our linting check or other related formatting requirements 

Here are some steps to take:

1. Click on the `Details` link for the tests to get an explanation of the cause(s)
2. You may have to fix a test that is not related to this PR as in some rare cases we merge faulty test code. If this is the case, fix the test in this PR. Do not open another issue.

We do this to ensure the overall reliability of the code base is maintained. This is a prerequisite for getting your PR approved. Assigned reviewers regularly review the PR queue and tend to focus on PRs that are passing.

Additional notes:

1. Ensure that CodeRabbit.ai approves your work
1. Please fix any conflicting files

File: src/components/CheckIn/tagTemplate.ts [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
-        characterSpacing: 0,
-        lineHeight: 1,
-        fontName: 'Roboto',
-        fontColor: '#08780b',
-      },
-    },
+  "schemas": [
+    [
+      {
+        "name": "xyz",
```
[the remark this answers is not in the document]
Comment (syedali237) [reply src/components/CheckIn/tagTemplate.ts:7]: Generate a type?

[the remark this answers is not in the document]
Comment (syedali237) [reply src/components/CheckIn/tagTemplate.ts:7]: @coderabbitai generate a type/interface for it?

[the remark this answers is not in the document]
Comment (syedali237) [reply src/components/CheckIn/tagTemplate.ts:7]: In the code below you can clearly observe, that a "name" does exist on the type Template. And running npm run typecheck on my local system it gives no error while here it is failing on Github   actions.

Screenshot 2024-12-22 at 12 38 57 AM

Screenshot 2024-12-22 at 12 39 09 AM

**Comment by syedali237** [comment]
Here, the linting check gives an error that the "name" property does not exist on the type Template, even though it's clearly defined as z.ZodString on the type Template. Running npm run typecheck locally doesn't show any issues, but it's failing on GitHub Actions.
Has anyone have suggestions regarding this?

Screenshot 2024-12-22 at 12 39 09 AM


Screenshot 2024-12-22 at 12 38 57 AM

**Comment by IITI-tushar** [comment]
In tagTemplate.ts try changing ` "name"  to name `  as given in here https://pdfme.com/docs/getting-started#template 
image

**Comment by syedali237** [comment]
@IITI-tushar Earlier it was "name" : "name" but still it was failing here.

**Comment by IITI-tushar** [comment]
@syedali237 try removing double quotes from name , schema , type....
From here

**Comment by syedali237** [comment]
@IITI-tushar this didn't work..

**Comment by IITI-tushar** [comment]
@syedali237 why have you make changes in tag templates?
Try to update the file as given in develop-postgres branch.

**Comment by syedali237** [comment]
> @syedali237 why have you make changes in tag templates? Try to update the file as given in develop-postgres branch.

@IITI-tushar it was giving an error while committing, I did mention that earlier. I tried committing the original file from the develop-postgres branch but still the same error arises. so had to fix it

**Comment by IITI-tushar** [comment]
@syedali237 I have recently made 2 prs and there was no error with tagtemplates
Try to run lint:fix and format:fix .

**Comment by syedali237** [comment]
> @syedali237 I have recently made 2 prs and there was no error with tagtemplates Try to run lint:fix and format:fix .

the file tagTemplate is already in .eslintignore, lint:fix doesn't cover it, but when typecheck is performed with the original tagTemplate.ts file from develop-postgres then it shows this error, otherwise with my changes it does not throw this error locally but here it fails on Github Actions.

Screenshot 2024-12-22 at 6 49 41 PM

**Comment by syedali237** [comment]
Closing this PR, had issues with my repository.

**Pull request closed without merging**